src/organisms/TopBar/Account/ImpersonateMenu/CreateNewUser/CreateNewUser.tsx

```diff
@@ -0,0 +1,122 @@
+import { ChangeEvent, FC, useMemo, useState } from 'react';
+
+import { Button, DotProgress, Icon, Typography } from '@equinor/eds-core-react';
+import { arrow_back } from '@equinor/eds-icons';
+import { AmplifyApplicationService } from '@equinor/subsurface-app-management';
+import { useQuery } from '@tanstack/react-query';
+
+import { useCreateImpersonation } from '../hooks/useCreateImpersonation';
+import { AVAILABLE_ROLES } from '../Impersonate.constants';
+import { Container, Header, Section } from './CreateNewUser.styles';
+import { environment } from 'src/atoms/utils/auth_environment';
+import { Switch } from 'src/molecules';
+import { TextField } from 'src/molecules/TextField/TextField';
+
+interface CreateNewUserProps {
+  onBack: () => void;
+  onClose: () => void;
+}
+
+export const CreateNewUser: FC<CreateNewUserProps> = ({ onBack, onClose }) => {
+  const [roles, setRoles] = useState<string[]>([]);
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+
+  const { data: availableRoles, isLoading: isLoadingRoles } = useQuery({
+    queryKey: [AVAILABLE_ROLES],
+    queryFn: () =>
+      AmplifyApplicationService.getAllAppRoles(
+        environment.getClientId(import.meta.env.VITE_CLIENT_ID)
+      ),
+  });
```

Comment:
This useQuery could be moved into its own hook in the hooks folder

Reply:
Pushed a new commit for this